source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -710,6 +710,16 @@ Why is waitQueueTimeoutMS optional for some drivers?
 
 We are anticipating eventually introducing a single client-side timeout mechanism, making us hesitant to introduce another granular timeout control. Therefore, if a driver/language already has an idiomatic way to implement their timeouts, they should leverage that mechanism over implementing waitQueueTimeoutMS.
 
+Why is there a distinction between a Pool being "closed" and the Pool resources being free-d / garbage collected? (a.k.a. why does a Pool have a "half-closed" state?)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are a few motivating factors for this decision:
+
+#. When a pool is closed, there may be connections belonging to the pool that are still checked out. In many situations, it would be very dangerous for the pool to aggressively close these checked-out connections. Therefore, **when a pool is closed, it does not aggresively close checked-out connections**.
+#. Blocking the driver on waiting for all connections to be closed by a pool could lead to performance hits, and in some cases deadlocks. Therefore, **when the driver closes a pool, it does not block on all connections being destroyed** (an exception may be made for application shutdown).
```

Comment:
I was mostly thinking of the second case. Is it mandated in the SDAM spec that the closing of connections in a pool must be done asynchronously? If so, then we can just reference that part of the spec here.

Reply:
Not sure if it's specified.  I know in Java driver that SDAM updates are made on the server monitor's thread, and since it has no more work to do we might as well use that thread to do all the socket close's before stopping the monitor.